vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -940,6 +940,78 @@ func (e *StandardTxExecutor) SetSubnetValidatorWeightTx(tx *txs.SetSubnetValidat
 	return nil
 }
 
+func (e *StandardTxExecutor) IncreaseBalanceTx(tx *txs.IncreaseBalanceTx) error {
+	var (
+		currentTimestamp = e.State.GetTimestamp()
+		upgrades         = e.Backend.Config.UpgradeConfig
+	)
+	if !upgrades.IsEtnaActivated(currentTimestamp) {
+		return errEtnaUpgradeNotActive
+	}
+
+	if err := e.Tx.SyntacticVerify(e.Ctx); err != nil {
+		return err
+	}
+
+	if err := avax.VerifyMemoFieldLength(tx.Memo, true /*=isDurangoActive*/); err != nil {
+		return err
+	}
+
+	// Verify the flowcheck
+	fee, err := e.FeeCalculator.CalculateFee(tx)
+	if err != nil {
+		return err
+	}
+
+	fee, err = safemath.Add(fee, tx.Balance)
+	if err != nil {
+		return err
+	}
+
+	if err := e.Backend.FlowChecker.VerifySpend(
+		tx,
+		e.State,
+		tx.Ins,
+		tx.Outs,
+		e.Tx.Creds,
+		map[ids.ID]uint64{
+			e.Ctx.AVAXAssetID: fee,
+		},
+	); err != nil {
+		return err
+	}
+
+	sov, err := e.State.GetSubnetOnlyValidator(tx.ValidationID)
+	if err != nil {
+		return err
+	}
+
+	// If the validator is currently inactive, we are activating it.
+	if sov.EndAccumulatedFee == 0 {
+		if gas.Gas(e.State.NumActiveSubnetOnlyValidators()) >= e.Backend.Config.ValidatorFeeConfig.Capacity {
```

Comment:
not sure I understand why `.Capacity` is a gas.Gas type?

Reply:
me neither but i also don't understand what we're comparing here at all - the left side is a number of validators, how can you cast it to a gas? and what's the capacity?

Reply:
Yes, this enforces the hard cap on the number of active validators as far as I can see

Reply:
This is kind of a chaining of the same reasoning over and over... We could change it if we wanted to...
- `Capacity` was chosen to be of `gas.Gas` so that it would have the same type as `Target`.
- `Target` was chosen to be of `gas.Gas` so that it would have the same type as `Excess`.
- `Excess` and `ExcessConversionConstant` were chosen to be of `gas.Gas` because the validator pricing is performed using the same calculations as ACP-103.

We could technically have `Capacity` and `Target` be specified as `int` for easier usage and have the casting be performed as an implementation detail of the fees package... Although then we might want to at least make them `uint` so that it is clear the casting can't overflow....

I felt like this was the cleanest... But open to suggestions (as a separate PR) to cleanup the types.